### PR TITLE
DHFPROD-4198:Create Save Query UI Component,DHFPROD-4628:Handle "save" and "save as" when there are unapplied changes

### PR DESCRIPTION
**Pull request by Sanjeevani19**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                        <Radio value={1}> Apply before saving</Radio>
+                        <Radio value={2}> Save as is, keep unapplied facets</Radio>
+                        <Radio value={3}> Discard unapplied facets</Radio>
+                    </Radio.Group>
+                </Form.Item>}
+                <Form.Item>
+                    <div className={styles.submitButtons}>
+                        <Button onClick={() => onCancel()}>Cancel</Button>
+                        &nbsp;&nbsp;
+                        <Button type="primary" htmlType="submit" onClick={()=> onOk()} id='save-query-button'>Save</Button>
```
Comment (timur-isangulov) [inline marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx:140]: Can you make Save button disabled if the query name input field is empty?

File: marklogic-data-hub-central/ui/src/assets/mock-data/query.js [hunk 2]
```diff
@@ -1,121 +1,159 @@
...
+            ]
+          }
         }
-    }]
+      },
+      "propertiesToDisplay": [
+        "facet1",
+        "EntityTypeProperty1"
+      ],
+      "owner": "admin",
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/assets/mock-data/query.js:33]: We are not supposed to be testing with admin. So can we remove/update all such references?

File: marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        if(queryName.length > 0 ){
+            props.saveNewQuery(queryName, queryDescription, facets);
+        } else {
+            isQueryEmpty('error')
+        }
+    }
+
+    const handleChange = (event) => {
+        if (event.target.id === 'save-query-name') {
+           setQueryName(event.target.value);
```
Comment (brucean52) [inline marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx:67]: I think there needs to be some basic validation for the name field. If save a query as an empty string, it doesn't show up in the save query dropdown, but it shows in the manage queries table.

File: marklogic-data-hub-central/ui/src/components/queries/saving/save-queries-dropdown/save-queries-dropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+
+interface Props {
+    savedQueryList: any[];
+    toggleApply: (clicked:boolean) => void;
+    greyFacets: any[];
+};
+
+
+const SaveQueriesDropdown: React.FC<Props> = (props) => {
```
Comment (brucean52) [inline marklogic-data-hub-central/ui/src/components/queries/saving/save-queries-dropdown/save-queries-dropdown.tsx:16]: I think the save query dropdown should be cleared whenever a search request is made. See examples.

https://drive.google.com/file/d/1lZWZGKOkRsbywOsd6R2W1A9H7JHAiWlH/view
https://drive.google.com/file/d/1BmWFT9ifZ45-ld3uit6sgoyTIs6Ip2Yy/view

File: marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx (hunk 3 above)
Comment (timur-isangulov) [reply marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx:67]: It shows on the save query dropdown and manage query for me. 
Screen Shot 2020-04-27 at 8 46 44 AM

File: marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+                        value={queryDescription}
+                        onChange={handleChange}
+                        placeholder={'Enter query description'}
+                    />
+                </Form.Item>
+                {props.greyFacets.length > 0  && <Form.Item
+                    colon={false}
+                    label='Unapplied Facets:'
+                    labelAlign="left"
+                    data-testid='unapplied-radio-options'
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx:128]: Same as above.

File: marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx (hunk 3 above)
Comment (Sanjeevani19) [reply marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx:67]: @brucean52 , @timur-isangulov, added the fixes.

File: marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx (hunk 5 above)
Comment (Sanjeevani19) [reply marklogic-data-hub-central/ui/src/components/queries/saving/save-query-modal/save-query-modal.tsx]: @bsrikan removed all the data-testid references

File: marklogic-data-hub-central/ui/src/components/queries/saving/save-queries-dropdown/save-queries-dropdown.tsx (hunk 4 above)
Comment (Sanjeevani19) [reply marklogic-data-hub-central/ui/src/components/queries/saving/save-queries-dropdown/save-queries-dropdown.tsx:16]: @brucean52 fixed this issue.

**Pull request merged**